plugins/search/providers/elastic_search/lib/model/search/ESearchEntryQueryFilterAttributes.php

```diff
@@ -34,13 +40,22 @@ public function getDisplayInSearchFilter()
 		{
 			return $mustNotDisplayInSearchBoolQuery;
 		}
-		
-		$displayInSearchBoolQuery = new kESearchBoolQuery();
-		$displayInSearchBoolQuery->addQueriesToShould($ignoreDisplayInSearchQueries);
-		if (!$queryContainsDisplayInSearch)
+
+		if (!$queryContainsDisplayInSearch || ($queryContainsDisplayInSearch && !$recycleDisplayInSearch))
```

Comment:
if (!$queryContainsDisplayInSearch || ($queryContainsDisplayInSearch && !$recycleDisplayInSearch))
equals to
 if (!$queryContainsDisplayInSearch ||  !$recycleDisplayInSearch))

remove redundant arg from if clause.